src/java/picard/sam/DownsampleSam.java

```diff
@@ -39,13 +48,21 @@
     @Option(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "The output, downsampled, SAM or BAM file to write.")
     public File OUTPUT;
 
+    @Option(shortName="S", doc="The downsampling strategy to use. See usage for discussion.")
+    public Strategy STRATEGY = Strategy.ConstantMemory;
+
     @Option(shortName = "R", doc = "Random seed to use if reproducibilty is desired.  " +
             "Setting to null will cause multiple invocations to produce different results.")
-    public Long RANDOM_SEED = 1L;
+    public Integer RANDOM_SEED = 1;
 
     @Option(shortName = "P", doc = "The probability of keeping any individual read, between 0 and 1.")
     public double PROBABILITY = 1;
```

Comment:
I think that @nh13 is talking about not running for p=1. I think that while
he is right that it's pointless to run in that case, I think that it should
be up to the user to optimize that away, not for the program to fail on
something it could (uselessly) do.

Y.

On Mon, Nov 9, 2015 at 11:42 AM, Tim Fennell <notifications@github.com>
wrote:

> In src/java/picard/sam/DownsampleSam.java
> <https://github.com/broadinstitute/picard/pull/334#discussion_r44297261>:
>
> >      @Option(shortName = "R", doc = "Random seed to use if reproducibilty is desired.  " +
> >              "Setting to null will cause multiple invocations to produce different results.")
> > -    public Long RANDOM_SEED = 1L;
> > +    public Integer RANDOM_SEED = 1;
> >
> >      @Option(shortName = "P", doc = "The probability of keeping any individual read, between 0 and 1.")
> >      public double PROBABILITY = 1;
>
> The DownsamplingIteratorFactory enforces that it is >=0 and <=1, and I
> didn't want to replicate here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/broadinstitute/picard/pull/334/files#r44297261>.
>